views/chat-view/src/components/ChatView/ChatView.tsx

```diff
@@ -66,13 +76,46 @@ export default function ChatView({
     formatter: profileFormatter
   });
 
+  useEffect(() => {
+    worker.current = new EmbedingWorker();
+
+    worker.current.onmessage = async (e) => {
+      if (e.data.type === "similarity") {
+        console.log("similarity: ", e.data.messages);
+        setSimilarityMessages(e.data.messages);
+        setShowSimilarityModal(true);
+        setSimilarityLoading(false);
+      }
+    }
+
+    return () => {
+      worker.current?.terminate();
+    };
+  }, [])
+
   async function submit() {
     try {
       const html = editor.current?.editor.getHTML();
+      const text = editor.current?.editor.getText();
       editor.current?.clear();
+      const queryEmbed = await queryEmbedding(text);
```

Comment:
Not sure if we should hard-code this into the Chat View. But I guess that is the merging with @jhweir's work that we discussed this morning?

Reply:
Yes, would be removed when we merge this with James work